Auto reopen previous documents

starofrainnight opened the pull request:
This PR just want to add feature which described in #135 .

It added an option "openLastFilesOnStartup" to control that behavior, it will disabled by default.

mitya57 commented on ReText/window.py:
```diff
@@ -353,6 +353,18 @@ def __init__(self, parent=None):
+
+			# Trigger the tab of last opened file
+			lastTabIndex = readFromSettings(
+				"lastTabIndex", int, default=self.tabWidget.currentIndex())
```
The `currentIndex` is (almost) always zero, so you can add `lastTabIndex` to the list of settings and use `globalSettings.lastTabIndex` instead.

mitya57 commented on ReText/window.py:
```diff
@@ -1078,6 +1090,13 @@ def closeEvent(self, closeevent):
+				tab = self.tabWidget.widget(i)
+				files.append(tab.fileName)
+			writeListToSettings("lastFileList", files)
+			writeToSettings("lastTabIndex", self.tabWidget.indexOf(self.currentTab), int)
```
… and `globalSettings.lastTabIndex = self.tabWidget.currentIndex()` here.

What you wrote is wrong because the third argument of `writeToSettings` is the default value, not the type.

Also, note that in my suggestion I replaced `indexOf()` with `currentIndex()` which is simplier.

starofrainnight commented:
@mitya57 

Already done your suggestions :

1. Moved lastTabIndex to  the list of settings and use globalSettings.lastTabIndex instead. So the writeToSettings() is unneeded now .
1. Replaced indexOf() with currentIndex()

mitya57 commented:
Pushed with minor changes and simplifications. Thanks!
